pkg/capr/planner/config.go

```diff
@@ -62,7 +62,14 @@ func (p *Planner) addETCD(config map[string]interface{}, controlPlane *rkev1.RKE
 			if v == "" {
 				config[k] = true
 			} else {
-				config[k] = v
+				// this part is specific for retention which is an int
+				if k == "etcd-s3-retention" {
+					if i, err := strconv.Atoi(v); err == nil {
```

Comment:
Suggesting to log warning if `Atoi` returned error - not just silently ignore error.